### ISSUE TITLE
Gravity Falls Legend of the Gnome Gemulets - Crashs With Shader JIT On

### ISSUE THREAD

**Issue by Kumoashi** [citra-emu/citra#2297]
I think I have found a problem with the Shader JIT.
I'm using Citra Master-6edb024 and the game is Gravity Falls Legend of the Gnome Gemulets.
If you try to play it with the Shader JIT on, Citra will crash.
![image](https://cloud.githubusercontent.com/assets/24490104/21073312/b82af1ae-bea8-11e6-8365-050fc8dc6b44.png)

Here is what the log in the console window says.
![image](https://cloud.githubusercontent.com/assets/24490104/21073326/e6ccbad8-bea8-11e6-854b-1b7c3da4a69d.png)

If you turn it off the game will start up and you can get in game, however, the game is still not playable, because all of the characters/enemies are invisible.
![image](https://cloud.githubusercontent.com/assets/24490104/21073333/1b760852-bea9-11e6-850e-287f259e16c5.png)

Turning the Shader JIT back on while the game is running will cause Citra to crash as well, but no errors are printed to the console window.
![image](https://cloud.githubusercontent.com/assets/24490104/21073336/42ace774-bea9-11e6-94aa-461897c96db1.png)

When I Debug Citra I get this.
![image](https://cloud.githubusercontent.com/assets/24490104/21073353/c137b9d4-bea9-11e6-99ee-a6143dc65a43.png)

I'm sorry if any of this is not list correctly. It's my first time posting an issue and I'm not completely sure how it's done.

**Comment by mailwl** [comment]
hmm... i remember i play this game without problem not so long time ago

**Comment by JayFoxRox** [comment]
This is also influenced by geometry shaders which probably would make the characters visible. Not sure about the JIT crash though.

**Comment by Kumoashi** [comment]
I can post the invisible characters as a separate issue if you like, but maybe I should wait until the crash is addressed first, to make sure that does have some sort of affect on it as well.

**Comment by Subv** [comment]
````RSP = 3```` yay for stack corruption

**Comment by wwylele** [comment]
Does this still happen on current master? If it does, then what if with #2520?

**Comment by mailwl** [comment]
now other place of crash:

![gravityfall](https://cloud.githubusercontent.com/assets/629437/22601950/47012112-ea52-11e6-8af8-854e828aebd2.png)

**Comment by wwylele** [comment]
@mailwl both master and #2520 crashes at the same place?

**Comment by mailwl** [comment]
yes, the same (memory addresses always differs, but code equals)

**Comment by Kumoashi** [comment]
The game seems to crash quicker now then before.

**Comment by JayFoxRox** [comment]
If you don't run this with GS instructions the flow could be all wrong..
Also by the nature of the jit, this is as much debugging output as you'll get.

If you want to contribute anything useful to this topic check if this is fixed with the proper GS flow (hard) or by at least dumping the faulty shader (using the interpreter and the pica shader debugger).

**Comment by mailwl** [comment]
@JayFoxRox may you write little tutorial, how to do it? It will be very useful

**Comment by Kumoashi** [comment]
@JayFoxRox I'm sorry, but I don't understand what most of that means. Could you explain it further, or give me a link that can help?

**Comment by Kumoashi** [comment]
Just out of curiosity, has there been any updates on fixing the Shader JIT crashes? I can still try to provide more info if someone can tell me how to get it.
I'm now running Citra on a much better laptop then I was before, so testing games doesn't take as long to do as before. I have VisualStudio up and running, but I haven't reinstalled CMake yet.

**Comment by Kumoashi** [comment]
I know the problem hasn't been fixed yet, but I figured that I would upload an updated copy of the crash log for this game in the hopes that it will help.

[log.txt](https://github.com/citra-emu/citra/files/1565147/log.txt)

**Comment by FearlessTobi** [comment]
I think this game has the same issue as https://github.com/citra-emu/citra/issues/3244. They are both Ubisoft titles and both crash before even showing the main menu. The only difference is that https://github.com/citra-emu/citra/issues/3244 shows some loading screens before crashing.

**Comment by Kumoashi** [comment]
Just a quick update.

Gravity Falls still crashes on latest Nightly build.
![image](https://user-images.githubusercontent.com/24490104/37246299-d17232a2-2474-11e8-8ee9-4615785d5c0e.png)

It, however, runs as intended on the latest Canary build.
![image](https://user-images.githubusercontent.com/24490104/37246281-ad944a14-2474-11e8-9123-e17d93afb7c7.png)

So, it seems that as soon as the new GPU related features are implemented into the Nightlys, then this issue can finally be closed.

**Comment by benderscruffy** [comment]
still happens in the latest nightly

**Comment by benderscruffy** [comment]
@FearlessTobi
can be closed
goes ingame with shader JIT enabled

**Comment by SleepingSnakezzz** [comment]
@benderscruffy Please test with shader JIT on, and Hardware Shaders disabled

**Comment by SleepingSnakezzz** [comment]
Nowadays, you wouldn't. However, back when this issue was first created, Hardware Shaders were not yet implemented. Regardless, this issue is about a crashing bug in our Shader JIT, which has not yet been resolved. Thus this should not be closed yet.
